torch/distributed/launch.py

```diff
@@ -249,14 +268,25 @@ def main():
 
         cmd.extend(args.training_script_args)
 
-        process = subprocess.Popen(cmd, env=current_env)
+        subprocess_stdout = None
+        if args.logdir:
+            directory_path = os.path.join(os.getcwd(), args.logdir)
+            file_handle = open(os.path.join(directory_path, "process_{}".format(local_rank)), "w")
```

Comment:
Would it make sense to also include the node_rank so that log file names from multiple machines do not collide?  Perhaps "process_{}_{}".format(node_rank, local_rank)?